gp-includes/routes/profile.php

```diff
@@ -16,17 +16,22 @@ public function profile_get() {
 		$this->tmpl( 'profile' );
 	}
 
-	public function profile_post() {
+	public function profile_post( $user_id = null ) {
 		if ( isset( $_POST['submit'] ) ) {
+			if ( (int)$user_id == 0 ) { 
```

Comment:
In my testing I found sometimes we may get a 0 passed in from WordPress, so the null check would not set the current user id correctly.